### PR TITLE
Gravity 5.0 status parsing refactor

### PR DESCRIPTION
**Pull request by wadells**
## Description
This is a refactoring of the code introduced in https://github.com/gravitational/robotest/pull/262.  All functionality is the same but this saves ~30 SLOC by removing the logic at:
```go
func (cs *ClusterStatus) UnmarshalJSON(data []byte) error
```
and replacing it with the more specific
```go
func (cs *SystemStatus) UnmarshalJSON(data []byte) error
```
(which previously mapped to `unmarshalSystemStatus`)

**_Why?_** While exploring @r0mant's comment here: https://github.com/gravitational/robotest/pull/262#discussion_r473265851, I found that using a dedicated type for SystemStatus simplified the custom parsing logic.  Furthermore, I grouped all status related code in `status.go` and `status_test.go` (previously distributed over `infra/gravity/node_commands.go`, `infra/gravity/gravity_test.go`, `lib/constants.go`, and `infra/gravity/parse.go`).

### Risk Profile
 - New feature or internal change (minor release)

### Related Issues
* Refactors the work done in https://github.com/gravitational/robotest/pull/262
* Touches https://github.com/gravitational/robotest/issues/247

## Testing Done
[5.0.36 install](https://console.cloud.google.com/logs/viewer?authuser=0&expandAll=false&project=kubeadm-167321&minLogLevel=0&timestamp=2020-08-24T18:34:22.692000000Z&customFacets=&limitCustomFacetWidth=true&advancedFilter=resource.type%3D%22project%22%0Alabels.__uuid__%3D%22fef5e858-5bc4-4c67-83ca-0496d29b5212%22%0Alabels.__suite__%3D%22754153be-c0c5-451c-9340-50a78bf9a22c%22%0Aseverity%3E%3DINFO&dateRangeEnd=2020-08-24T18:34:19.625Z&interval=PT1H&dateRangeUnbound=backwardInTime&scrollTimestamp=2020-08-22T02:16:03.966309118Z)

<details><summary><code>make test</code></summary>

```
walt@work:~/git/robotest$ go test ./infra/gravity/...
ok      github.com/gravitational/robotest/infra/gravity 0.014s
walt@work:~/git/robotest$ make test
docker run --rm=true -u $(id -u):$(id -g) -v /home/walt/git/robotest:/go/src/github.com/gravitational/robotest -v /home/walt/git/robotest/build:/go/src/github.com/gravitational/robotest/build -w /go/src/github.com/gravitational/robotest \
        --env="GO111MODULE=off" \
        robotest:buildbox \
        dumb-init go test -cover -race -v ./infra/... ./lib/config/...
=== RUN   TestCreatesPool
--- PASS: TestCreatesPool (0.00s)
=== RUN   TestAllocatesAndFrees
--- PASS: TestAllocatesAndFrees (0.00s)
=== RUN   TestFailsToAllocBeyondCapacity
--- PASS: TestFailsToAllocBeyondCapacity (0.00s)
=== RUN   TestDoesnotFreeNonExisting
--- PASS: TestDoesnotFreeNonExisting (0.00s)
PASS
coverage: 7.1% of statements
ok      github.com/gravitational/robotest/infra 0.067s  coverage: 7.1% of statements
=== RUN   TestParseClusterStatus
=== RUN   TestParseClusterStatus/NotFound
=== RUN   TestParseClusterStatus/Status
--- PASS: TestParseClusterStatus (0.00s)
    --- PASS: TestParseClusterStatus/NotFound (0.00s)
    --- PASS: TestParseClusterStatus/Status (0.00s)
=== RUN   TestDDOutputParser
--- PASS: TestDDOutputParser (0.00s)
=== RUN   TestGravityOutput
--- PASS: TestGravityOutput (0.00s)
=== RUN   TestHealthyStatusValidation
--- PASS: TestHealthyStatusValidation (0.00s)
=== RUN   Test1523StatusValidation
--- PASS: Test1523StatusValidation (0.00s)
=== RUN   Test1641StatusValidation
--- PASS: Test1641StatusValidation (0.00s)
=== RUN   TestGravity5036ActiveStatusValidation
--- PASS: TestGravity5036ActiveStatusValidation (0.00s)
=== RUN   TestGravity5036DegradedStatusValidation
--- PASS: TestGravity5036DegradedStatusValidation (0.00s)
=== RUN   TestSystemStatusStringDegradedUnmarshal
--- PASS: TestSystemStatusStringDegradedUnmarshal (0.00s)
=== RUN   TestSystemStatusStringRunningUnmarshal
--- PASS: TestSystemStatusStringRunningUnmarshal (0.00s)
=== RUN   TestSystemStatusIntUnmarshal
--- PASS: TestSystemStatusIntUnmarshal (0.00s)
=== RUN   TestSystemStatusInvalidFloatUnmarshal
--- PASS: TestSystemStatusInvalidFloatUnmarshal (0.00s)
=== RUN   TestSystemStatusInvalidStringUnmarshal
--- PASS: TestSystemStatusInvalidStringUnmarshal (0.00s)
PASS
coverage: 3.3% of statements
ok      github.com/gravitational/robotest/infra/gravity 0.104s  coverage: 3.3% of statements
?       github.com/gravitational/robotest/infra/providers/aws   [no test files]
?       github.com/gravitational/robotest/infra/providers/azure [no test files]
?       github.com/gravitational/robotest/infra/providers/gce   [no test files]
?       github.com/gravitational/robotest/infra/providers/ops   [no test files]
=== RUN   TestConvertConfigToTerraformVars
--- PASS: TestConvertConfigToTerraformVars (0.00s)
=== RUN   TestConvertConfigToTerraformVarsOptionalValues
--- PASS: TestConvertConfigToTerraformVarsOptionalValues (0.00s)
PASS
coverage: 10.0% of statements
ok      github.com/gravitational/robotest/infra/terraform       0.049s  coverage: 10.0% of statements
=== RUN   TestParsesSSHConfig
time="2020-08-24T18:31:11Z" level=info msg="obtained: []infra.Node{(*vagrant.node)(0xc00000ef40)}"
time="2020-08-24T18:31:11Z" level=info msg="obtained: []infra.Node{(*vagrant.node)(0xc00000f000)}"
--- PASS: TestParsesSSHConfig (0.00s)
PASS
coverage: 7.8% of statements
ok      github.com/gravitational/robotest/infra/vagrant 0.031s  coverage: 7.8% of statements
=== RUN   TestCompleteParamValidation
--- PASS: TestCompleteParamValidation (0.00s)
=== RUN   TestPartialParamValidation
--- PASS: TestPartialParamValidation (0.00s)
=== RUN   TestInvalidUIDParamValidation
--- PASS: TestInvalidUIDParamValidation (0.00s)
=== RUN   TestNilableNestedParamDefault
--- PASS: TestNilableNestedParamDefault (0.00s)
=== RUN   TestMakeFunction
{"uid":0,"user":"root","operation":{"name":"stop","timeout":"5s"}}
--- PASS: TestMakeFunction (0.00s)
=== RUN   TestMakeFunctionNoData
--- PASS: TestMakeFunctionNoData (0.00s)
=== RUN   TestJsonDeserialization
--- PASS: TestJsonDeserialization (0.00s)
=== RUN   TestInvalidDeserialization
--- PASS: TestInvalidDeserialization (0.00s)
=== RUN   TestDurationSerialization
--- PASS: TestDurationSerialization (0.00s)
PASS
coverage: 38.1% of statements
ok      github.com/gravitational/robotest/lib/config    0.083s  coverage: 38.1% of statements
```
</details>

### CODE REVIEW

File: infra/gravity/status.go [hunk 1]
```diff
@@ -17,30 +17,141 @@ limitations under the License.
...
+	ClusterStateActive = "active"
+	// ClusterStateDegraded is unhealthy.
+	ClusterStateDegraded = "degraded"
+)
+
+// SystemStatus comes from https://github.com/gravitational/satellite/blob/7.1.0/agent/proto/agentpb/agent.pb.go#L28-L32
+type SystemStatus int
+
+const (
+	SystemStatus_Unknown  SystemStatus = 0
```
Comment (r0mant) [inline infra/gravity/status.go:45]: Nit: All top-level exported names should have Godocs on them.

File: infra/gravity/status.go [hunk 2]
```diff
@@ -17,30 +17,141 @@ limitations under the License.
...
+var SystemStatusNames = [...]string{"undefined", "running", "degraded"}
+
+func (ss SystemStatus) String() string {
+	return SystemStatusNames[ss]
+}
+
+// UnmarshalJSON fulfills the Unmarshaler interface, allowing SystemStatus
+// to perform custom JSON unmarshalling.
+//
+// Gravity 5.2.x+ marshals SystemStatus as an integer and Gravity 5.0.36-
```
Comment (r0mant) [inline infra/gravity/status.go:59]: I would prefix this part with `TODO` so this potentially can be searched for later.

File: infra/gravity/status.go (hunk 1 above)
Comment (wadells) [reply infra/gravity/status.go]: Documented in 3051902.

File: infra/gravity/status.go (hunk 2 above)
Comment (wadells) [reply infra/gravity/status.go:62]: Added a TODO in 3051902.

**Pull request merged**